### DISCUSSION TITLE
xDrip Sync - what you need to know about updating

### DISCUSSION THREAD

**Discussion by jamorham** [NightscoutFoundation/xDrip#3664] [Announcements]
Around 20th July 2024, Google discontinued support for a key messaging feature that xDrip used for the sync feature. They kindly granted us an extension which is due to expire on 20th September.

**If you are using an xDrip version later than 23rd July then you will already be using the replacement system that I put in place to bridge the gap of this missing feature and you do not need to do anything.  After 20th September it is expected that xDrip sync will stop working on all previous versions of the app. If you are using an older version and using xDrip sync then you need to update your xDrip app.** 

In addition to this issue, the alternative Firebase messaging facility which we are now using is designed with a much narrower vision of what apps can do and ultimately I don't think it will support our use case long term.

As such, I have put together a replacement system for both Firebase and Google Cloud Messaging that is called **xDrip Cloud**. Long-term this is intended to replace the Google facilities we were using and should give much more flexibility as I've specifically designed it for xDrip's use case. The xDrip Cloud feature is present in xDrip versions starting from 15th September.

If you find that the the new xDrip sync present in 23rd July release and onwards doesn't work for you - for example some cellular carriers appear to block it, then please try enabling the "**xDrip Cloud**" feature in the "**xDrip+ Sync Settings**" settings screen. Make sure that all handsets in your follow group have this setting set the same.

Please report back any successes or failures using the new xDrip Cloud feature that you think are relevant. It is a new feature so may take a little while before it as smooth and reliable as I would like.

Please also note that whichever version of xDrip Sync you are using, all data is, and always has been, end-to-end encrypted between handsets and there is no way that I or anyone else can see things like your glucose numbers etc.

**Comment by vincepf** [comment]
Thank you @jamorham for all of your tremendous work in behalf of others.

**Comment by parapenT1sta** [comment]
Thank you so much for all your work!

Just one question If the xDrip Cloud is enabled on the master phone, will this setting also be set to "Enabled" on the follower phones that use QR code to set them as followers? or we have to set it manually?

**Reply by Navid200** [reply]
The instructions for using the QR code to set a follower asks you to show the master's xDrip security keys only.  
Scanning that QR code on the follower will not set or reset the Use xDrip Cloud setting on the follower.  

You need to set or reset that setting manually to match the master.  
https://navid200.github.io/xDrip/docs/xDripSync.html

**Reply by Navid200** [reply]
That is an interesting view.  It may be possible to add that feature.  I'm not sure.  
My limited understanding of the whole thing is that eventually it should be enabled for all.  But, since this is new now, the setting may allow one to get operation while we troubleshoot the main (xDrip cloud) path.

**Reply by parapenT1sta** [reply]
I just asked because as I know this option must be enabled on master and followers phones.

**Reply by Navid200** [reply]
I know.

We should be able to modify xDrip so that when you ask xDrip master to show the security keys only, it would also include the xDrip Cloud setting.  Then, as you suggested, scanning the QR code by the follower would also set or unset the xDrip Cloud setting to match the master setting.

The rest of what I said was meant to say that I am not sure it is worth the trouble because eventually there should be no need for the setting and hopefully, it will be force set with the setting removed from the UI.  But, who knows when we will get to that point.
